src/compile/mark/encode/tooltip.ts

```diff
@@ -164,3 +165,23 @@ export function tooltipRefForEncoding(
   const keyValues = entries(data).map(([key, value]) => `"${key}": ${value}`);
   return keyValues.length > 0 ? {signal: `{${keyValues.join(', ')}}`} : undefined;
 }
+
+/**
+ * Transforms a tooltip value that is an array to a string with line breaks
+ */
+function addLineBreaksToTooltip(
+  channelDef: Encoding<string>['text' | 'tooltip'],
+  config: Config,
+  expr: 'datum' | 'datum.datum' = 'datum',
+): VgValueRef {
+  // tooltip fields with a format property are no strings
+  const fieldDefWithFormat = channelDef as {field: string; type: string; format: string};
```

Comment:
how do you know that the type here is `{field: string; type: string; format: string}`?

Reply:
I'm still wrestling a bit with the elaborate type system in this repo 😅
I have replaced the cast with a type guard.